src/dune/dpath.ml

```diff
@@ -35,14 +36,21 @@ let analyse_target (fn as original_fn) : target_kind =
   | Some ("install", sub) -> (
     match Path.Local.split_first_component sub with
     | None -> Other fn
-    | Some (ctx, fn) -> Install (ctx, Path.Source.of_local fn) )
-  | Some (ctx, sub) -> Regular (ctx, Path.Source.of_local sub)
+    | Some (ctx, fn) ->
+      let ctx = Context_name.of_string ctx in
```

Comment:
Should a failure here also lead to "Other"?

Reply:
The other clause was changed to `Other` because it would break dune otherwise. I was going to ask you regarding that actually. 

Here, I think we should just error out on an invalid context name as it can't occur unless the user messed up. I'll change it to `parse_string_exn` to give a proper message.

Reply:
I think this low-level utility function shouldn't try to report a good error message. If the distinction between Other and Invalid is important, we could introduce it, but I suspect just returning "Other" might be all right (hard to know without inspecting call sites).

By the way, I think this failure is equally likely to indicate dune rules bug as user error.

In any case, if after reading this you still disagree then I think it's all right to keep it as is.

Reply:
You said you wanted to ask me about that other `Other`, but I have no thoughts about it. :-) If you have a more specific question, we could discuss that.

Reply:
For some reason we were trying to turn `"."` into a context in this clause (old code):
```ocaml
  | Some (ctx, sub) -> Regular (ctx, Path.Source.of_local sub)
```

Here ctx is `"."`.

In any case, let's keep this function higher level as you said. I turned the other clause to `Other ..` too.

Reply:
I thought this was weird enough that it's worth investigating.
Look at what I've found:

```
  let extract_build_context t =
    let t = Local.to_string t in
    match String.lsplit2 t ~on:'/' with
    | None -> Some (t, Source0.root)
    | Some (before, after) -> Some (before, after |> Source0.of_string)
```

When do you think this function returns `None`? :-)
I'm going to go make a fix.